### ISSUE TITLE
Vehicles Concept

### ISSUE THREAD

**Issue by MajorLunaC** [iamgreaser/iceball#145]
This is just some concept for the distant future, that might attract more people someday. You're probably laughing or angry at how ridiculous it seems at this point, I may have a way to simplify vehicle implementation a bit. Really it's just something amusing to think about.

I've noticed that some games have actually gotten away with mostly static, non-tilting vehicles. Basically, vehicles act exactly like a player, basically replacing the player model, have fewer weapons, and can't jump. I've noticed that other games use the highest point of ground collision with the vehicle to position the vehicle, simply lifting the vehicle up and having it move forward with just a single edge toughing the ground. It's pretty much one big rectangle (collision box) that can only rotate around the y-axis (turn a vehicle left or right). It just moves along every other axis, without rotating. Here's a visual in a high-definition 3D game (minute 4:00+): https://www.youtube.com/watch?v=0uGBBtVUYhY

Maybe if the leg area on a player model were stretched into a large rectangle, it be a start in the movement. More health for a vehicle. As for weapons say for a tank, grenades that explode on impact, SMG, and possibly 1-block or 2-block range shotgun to serve as a drilling machine. The only problem is building is so slow and limited at the moment that the map will be flat in no time, so maybe build more blocks at one time? Hmmm...

**Comment by Dany0** [comment]
tl;dr

I wrote a hackish mod for the hackish third person cam mod with custom obj_player that has car.pmf. They need custom physics. Might commit if I could be arsed. Oh it broke with new model_load. welp. \o/

**Comment by rakiru** [comment]
It could be implemented as a mod in the future. How it's done would depend entirely on who writes it, if anyone.

I'm mostly replying for your final point though. We are thinking of ways to allow quicker building, and welcome any input on that matter.

**Comment by MajorLunaC** [comment]
Oh, for quicker building, so far I've seen what build-and-shoot has, which allows you to drag-and-select an area to fill in with blocks, but building in the air is a slight issue (need framework first, like a pole). Or what some minecraft clones use, which is selecting a source block and a destination block and then auto-filling in-between. Building more rapidly is also an issue of:
*Building range: building across a gorge is a nightmare.
*Building in the air: Selecting a position to build in the air for auto-filling of blocks to that point. It might help to use the mouse or keys to select a position in the x and y plane (as if part of an invisible wall) and have a couple of keys to shift along the z plane/depth. Basically closer to an editor, than manual building. Slow-speed zip-lines (source block + destination block; walking speed) and ropes/ladders could help too, but that might be pushing far beyond building.
*Block supply: To build something that won't disappear in 1 shot, you have to build large reinforced structures, but you just never seem to have enough blocks. Full reload of blocks should happen at your base flag, but you should also be able to dig faster, especially since the maps seem much bigger (I think). Total inventory supply should be greater to begin with, especially what you start out with. Maybe even being able to drop a backpack full of blocks (ally helping out to supply from base, or dead enemy).
*Thinking it's worthwhile building something: I don't like the concept of AoS and build-and-shoot for destroying blocks, because your amazing tower or bridge that took hours to build can be destroyed in seconds, and there wont be anything left because it all magically disappears. I would at least prefer for the blocks to just fall straight down like sand when they're disconnected, maybe even reconnect at the lowest point of contact, instead of disappearing. This would also make some interesting ceiling-crush traps and floor-fall-away traps.

I'm sure I can come up with more ideas and solutions if you really want, but I think these are just about all of the most practical ones.

**Comment by rakiru** [comment]
Closing as not really suited to an issue tracker. A better place to discuss potential future mechanics is the subreddit: http://www.reddit.com/r/buildandshoot
